src/radix_connect/app_link_parser.rs

```diff
@@ -0,0 +1,154 @@
+use crate::prelude::*;
+use url::form_urlencoded;
+use url::Url;
+
+use super::{DappRequest, LinkRequest, MobileConnectRequest};
+
+#[uniffi::export]
+pub fn parse_request_url(url: String) -> Result<MobileConnectRequest> {
```

Comment:
oh also, you violate an important principle here, our UniFFI exported method should NEVER have logic in them, they should ALWAYS just contain one single trivial line calling another, non UniFFI exported function.

So change to let `MobileConnectRequest` impl `FromStr` and use it:

(inside the 
```rust
impl FromStr for MobileConnectRequest {
    type Err = CommonError;
    fn from_str(s: &str) -> Result<Self> { /* this function below... */}
}
```

and then this UniFFI exported function should be just:

```rust
#[uniffi::export]
pub fn new_mobile_connect_request(url: String) -> Result<MobileConnectRequest> {
    MobileConnectRequest::from_str(&url)
}

#[cfg(test)]
mod uniffi_test { 
    use super::*;

    #[allow(clippy::upper_case_acronyms)]
    type SUT = MobileConnectRequest;
   #[test]
   fn test_new_mobile_connect_request() {
        assert!(new_mobile_connect_request("some example").is_ok()) // at least this test
   }
}
```